test/unit/BPool/BPool_Bind.t.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {BPoolBase} from './BPoolBase.sol';
+import {IBPool} from 'interfaces/IBPool.sol';
+
+contract BPool_Bind is BPoolBase {
+  function test_WhenReentrancyLockIsNOTSetRevertWhen_CallerIsNOTController(address _caller) external {
```

Comment:
had to walk a bit around bulloak's inconsistencies and only managed to flatten the tree when the 'default' (ie: not doing anything) is the required setup, since checks are executed one after the other in the code